agent/src/main/java/com/teamscale/jacoco/agent/options/AgentOptions.java

```diff
@@ -225,9 +229,15 @@ public String getOriginalOptionsString() {
 				"You specified additional meta data files to be uploaded but did not configure an upload URL");
 
 		validator.isTrue(teamscaleServer.hasAllRequiredFieldsNull() || teamscaleServer
-						.hasAllRequiredFieldsSet() || sapNetWeaverJavaApplications != null,
+						.hasAllRequiredFieldsSetExceptProject() || sapNetWeaverJavaApplications != null,
 				"You did provide some options prefixed with 'teamscale-', but not all required ones!");
 
+		validator.isFalse(teamscaleServer.hasAllRequiredFieldsSetAndProjectNull() && (teamscaleServer.revision != null
+				|| teamscaleServer.commit != null),"You tried to provide a commit to upload to directly. This is not possible, since you" +
```

Comment:
Please add a test case to ensure that the expected combinations of allowed and disallowed values are rejected/accepted (See `AgentOptionsTest`)